duckduckgo/iOS
# 23. Subscriptions:  - Feature Flags

Task/Issue URL: https://app.asana.com/0/0/1206807550798011/f
Plan: https://app.asana.com/0/0/1206820386373649/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/717/files

**Description**:
- Adds feature flags for Subscriptions availability (Core logic in BSK)
- Adds compiler flags for Subscription environment

**How to test (Feature Availability)**:
- Build DDG-Alpha target
- Confirm Subscription is available in settings
- Go to Debug Settings > UnCheck Internal User State
- Confirm Subscription is NOT available in settings

**How to test (Disallowing Purchases)**:
- Invert conditional in `SubscriptionPagesUseSubscriptionFeature > getSubscriptionOptions()` so it returns an `SubscriptionOptions.empty`
- Go to `Settings > Privacy Pro > Get Privacy Pro`
- Confirm Purchase buttons are not visible. (See image below)

**Optional:**
Test with a custom privacy config URL > Flags available in: https://github.com/duckduckgo/privacy-configuration/pull/1887

![Screenshot 2024-03-16 at 1 18 35 AM](https://github.com/duckduckgo/iOS/assets/1156669/ec3d95f0-19ce-4dd6-a13b-2054b60a2afa)
